testing/trino-faulttolerant-tests/src/test/java/io/trino/faulttolerant/BaseFaultTolerantExecutionTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.faulttolerant;
+
+import io.trino.Session;
+import io.trino.testing.AbstractTestQueryFramework;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+
+public abstract class BaseFaultTolerantExecutionTest
+        extends AbstractTestQueryFramework
+{
+    private final String partitioningTablePropertyName;
+
+    protected BaseFaultTolerantExecutionTest(String partitioningTablePropertyName)
+    {
+        this.partitioningTablePropertyName = requireNonNull(partitioningTablePropertyName, "partitioningTablePropertyName is null");
+    }
+
+    @Test
+    public void testTableWritePreferredWritePartitioningSkewMitigation()
+    {
+        // We use TEXTFILE in this test because is has a very consistent and predictable size
```

Comment:
nit: is -> it

Reply:
We actually don't use text file in this test. A copy-paste mistake